--- server-config.wsdd.orig	2007-02-01 09:31:34.000000000 +1000
+++ server-config.wsdd	2007-02-01 09:37:45.000000000 +1000
@@ -68,18 +68,20 @@
 
     <service name="DefaultIndexService" provider="Handler" 
         use="literal" style="document">
         <parameter name="providers" 
                    value="org.globus.wsrf.impl.servicegroup.ServiceGroupRegistrationProvider 
+                          org.globus.mds.usefulrp.rpprovider.ResourcePropertyProviderCollection
                           GetRPProvider
                           GetMRPProvider 
                           QueryRPProvider
                           DestroyProvider 
                           SetTerminationTimeProvider 
                           SubscribeProvider 
                           GetCurrentMessageProvider"/>
 
+        <parameter name="rpProviderConfigFile" value="etc/globus_wsrf_mds_index/gluece-rpprovider-config.xml"/>
         <parameter name="handlerClass" value="org.globus.axis.providers.RPCProvider"/>
         <parameter name="scope" value="Application"/> 
         <parameter name="allowedMethods" value="*"/>
         <parameter name="className" 
             value="org.globus.mds.index.impl.DefaultIndexService"/>
