--- iRODS/lib/core/src/sockComm.c.orig	2010-10-06 15:31:00.000000000 +1300
+++ iRODS/lib/core/src/sockComm.c	2010-10-06 16:41:48.813851911 +1300
@@ -771,7 +771,7 @@
 	    } else if (status > 0) {
 		int myval;
 #if defined(aix_platform)
-        	size_t mylen = sizeof (int);
+        	socklen_t mylen = sizeof (int);
 #else
 		uint mylen = sizeof (int);
 #endif
@@ -860,7 +860,7 @@
 setRemoteAddr (int sock, struct sockaddr_in *remoteAddr)
 {
 #if defined(aix_platform)
-    size_t      laddrlen = sizeof(struct sockaddr);
+    socklen_t      laddrlen = sizeof(struct sockaddr);
 #elif defined(windows_platform)
 	int laddrlen = sizeof(struct sockaddr);
 #else
@@ -885,7 +885,7 @@
 setLocalAddr (int sock, struct sockaddr_in *localAddr)
 {
 #if defined(aix_platform)
-    size_t      laddrlen = sizeof(struct sockaddr);
+    socklen_t      laddrlen = sizeof(struct sockaddr);
 #elif defined(windows_platform)
     int         laddrlen = sizeof(struct sockaddr);
 #else
