--- install_mip.orig	2006-11-30 06:39:00.000000000 +1000
+++ install_mip	2006-12-18 14:15:40.000000000 +1000
@@ -10,12 +10,12 @@
 my $mipdir=getcwd;
 my $vdt_location=$ENV{VDT_LOCATION};
 my $defaultproducer='glue';
-my @pkgs=('osg','int','site');
+my @pkgs=('apac_py');
 
 
 if (@ARGV < 1){
    install_mip($mipdir, $vdt_location, $defaultproducer, @pkgs);
-   print "\n Please edit 'config/osg-conf.pl' and set additional parameters\n";
+#   print "\n Please edit 'config/osg-conf.pl' and set additional parameters\n";
 }
 else{
    install_pkg($ARGV[0]);
--- lib/installer.pm.orig	2006-12-05 01:38:10.000000000 +1000
+++ lib/installer.pm	2006-12-18 14:24:43.000000000 +1000
@@ -18,7 +18,7 @@
 sub install_mip{
 	my($mipdir, $vdt_location,$defaultproducer, @pkgs) = @_;
    die "Please cd to MIP directory then run ./install_mip" if not -e "$mipdir/mip.pl";
-   die "Please set VDT_LOCATION before installing MIP" if not -d "$vdt_location";
+#   die "Please set VDT_LOCATION before installing MIP" if not -d "$vdt_location";
    createsource($mipdir,"$mipdir/config","$mipdir/modules",$defaultproducer,@pkgs);
    createmip($mipdir,"$mipdir/config",$vdt_location);
 }
@@ -49,7 +49,8 @@
 	open(outfile, ">$mipdir/mip") or die "cannot write 'mip'";
 	print outfile "#!/bin/bash\n" .
 		"LANG=C\n" .
-		". $vdt_location/setup.sh\n" .
+#		". $vdt_location/setup.sh\n" .
+		"export PYTHONPATH=\"$mipdir/modules/apac_py:\$PYTHONPATH\"\n" .
 		"cd $mipdir\n" .
 		"if [ ! -z \"\$1\" ]; then\n" .
 		"   if [ \"\$1\" == \"-remote\" ]; then\n" .
--- lib/functions.pm.orig	2007-01-30 22:29:13.000000000 +1000
+++ lib/functions.pm	2007-01-30 22:30:46.000000000 +1000
@@ -42,6 +42,9 @@
 {						# These would normally hash to the same value
 	my ($el)=@_;   # Need to add all elements that will have same Attribute for multiple values
 	switch ($el) {
+		case /^Sponsor/ { return 1; }
+		case /^OtherInfo/ { return 1; }
+		case /^ContactString/ { return 1; }
 		case /^Rule/ { return 1; }
 		case /^Variable/ { return 2; }
 		case /^GlueCEAccessControlBaseRule/ { return 3; }
@@ -228,12 +231,14 @@
 		case /^ComputingElement/ { @arr=('InformationServiceURL','Name','LRMSType','LRMSVersion','GRAMVersion','HostName','GateKeeperPort','JobManager','ContactString',
 											'ApplicationDir','DataDir','DefaultSE','Status','RunningJobs','WaitingJobs','TotalJobs','EstimatedResponseTime','WorstResponseTime',
 											'FreeJobSlots','MaxWallClockTime','MaxCPUTime','MaxTotalJobs','MaxRunningJobs','Priority','AssignedJobSlots','ACL','Job','VOView'); }
-		case /^SubCluster/ { @arr=('Name','PhysicalCPUs','LocalCPUs','TmpDir','WNTmpDir','OperatingSystem','Processor','NetworkAdapter','MainMemory','Architecture',
-									'Benchmark','Location','RunTimeEnv',); }
+		case /^SoftwarePackage/ {  @arr=('Name','Version','Path','QueueResource','Module','SerialAvail','ParallelAvail','ParallelMaxCPUs','SoftwareExecutable'); }
+		case /^SoftwareExecutable/ {  @arr=('Name','Version','Path','SerialAvail','ParallelAvail','ParallelMaxCPUs'); }
+		case /^SubCluster/ { @arr=('Name','PhysicalCPUs','LogicalCPUs','TmpDir','WNTmpDir','OperatingSystem','Processor','NetworkAdapter','MainMemory','Architecture',
+									'Benchmark','Location','RunTimeEnv','SoftwarePackage'); }
 		case /^StorageArea/ { @arr=('Path','Type','Quota','MinFileSize','MaxFileSize','MaxData','MaxNumFiles','MaxPinDuration','UsedSpace','AvailableSpace','ACL'); }
 		case /^Cluster/ { @arr=('Name','TmpDir','WNTmpDir','ComputingElement','SubCluster'); }
 		case /^Service/ { @arr=('Name','Type','Version','Endpoint','Status','StatusInfo','WSDL','Semantics','StartTime','Owner','Data'); }
-		case /^StorageElement/ { @arr=('InformationServiceURL','SizeTotal','SizeFree','Architecture','StorageArea','AccessProtocol','ControlProtocol'); }
+		case /^StorageElement/ { @arr=('Name','InformationServiceURL','SizeTotal','SizeFree','Architecture','StorageArea','AccessProtocol','ControlProtocol'); }
 		case /^Host/ { @arr=('Name','UpTime','Architecture','MainMemory','OperatingSystem','Processor','Load','NetworkAdapter','Benchmark','RunTimeEnv','StorageDevice',
 									'StoragePartition','LocalFileSystem','RemoteFileSystem','StorageDevice2StoragePartition','StoragePartition2FileSystem'); }
 		case /^Site/ { @arr=('Name','Description','UserSupportContact','SysAdminContact','SecurityContact','Location','Latitude','Longitude','Web','Sponsor','OtherInfo',
