--- ./lib/core/src/miscUtil.c.bak	2010-02-10 12:53:45.000000000 +1100
+++ ./lib/core/src/miscUtil.c	2010-05-14 10:41:38.000000000 +1000
@@ -771,6 +771,10 @@
       free (dataObjSqlResult->ownerName.value);
     if (dataObjSqlResult->replNum.value != NULL)
       free (dataObjSqlResult->replNum.value);
+    if (dataObjSqlResult->dataMode.value != NULL)
+        free (dataObjSqlResult->dataMode.value);
+    if (dataObjSqlResult->rescGrp.value != NULL)
+        free (dataObjSqlResult->rescGrp.value);
 
     memset (dataObjSqlResult, 0, sizeof (dataObjSqlResult_t));
 
